### ISSUE TITLE
Installing with Wine 3.0 : Sorta works, then fonts don't load on restart.

### ISSUE THREAD

**Issue by fdrobidoux** [Nexus-Mods/Vortex#93]
**Disclaimer**: I know this is an alpha and Linux/Wine support is way out of scope, but I still feel it's interesting to point out the problems occurring with launching the program with Wine. Also, if you don't know anything about Wine, this isn't for you. I also have developed with Electron (the nodejs native app framework used to make Vortex) and I know it is possible to make some of this work natively on linux, so this is my way to expose the current status on UNIX-based platforms.

I've somehow been able to install Vortex in a wine prefix I made specifically to play Fallout 4. On the first install, it crashed, and I had to change my wineconfigs to Windows 7 in order to make it launch. And surprisingly, it launched no problem!

...until I restarted the application. It now won't load the fonts nor most images, so most text is blank. This also occurs in other wineprefixes.

Even worse, some wineprefixes make it crash instantly, mentioning this in the details : 

```
0x0000000140412f99: int	$3
Modules:
Module	Address					Debug info	Name (172 modules)
PE	          240000-          5df000	Deferred        ffmpeg
ELF	        7a800000-        7a9dd000	Deferred        opengl32<elf>
  \-PE	        7a850000-        7a9dd000	\               opengl32
ELF	        7b400000-        7b80e000	Deferred        kernel32<elf>
  \-PE	        7b420000-        7b80e000	\               kernel32
ELF	        7bc00000-        7bd16000	Deferred        ntdll<elf>
  \-PE	        7bc20000-        7bd16000	\               ntdll
ELF	        7c000000-        7c004000	Deferred        <wine-loader>
PE	       140000000-       144d52000	Export          vortex
PE	       180000000-       1811e6000	Deferred        node
ELF	    7fecb7a33000-    7fecb7d00000	Deferred        msvcrt<elf>
  \-PE	    7fecb7a50000-    7fecb7d00000	\               msvcrt
ELF	    7fecbe177000-    7fecbe38e000	Deferred        powrprof<elf>
  \-PE	    7fecbe180000-    7fecbe38e000	\               powrprof
ELF	    7fecbe38e000-    7fecbe600000	Deferred        libpcre.so.3
ELF	    7fecbe600000-    7fecbe815000	Deferred        libgpg-error.so.0
ELF	    7fecbe815000-    7fecbea3d000	Deferred        libselinux.so.1
ELF	    7fecbea3d000-    7fecbec55000	Deferred        liblz4.so.1
ELF	    7fecbec55000-    7fecbee7b000	Deferred        liblzma.so.5
ELF	    7fecbee7b000-    7fecbf189000	Deferred        libgcrypt.so.20
ELF	    7fecbf189000-    7fecbf40e000	Deferred        libsystemd.so.0
ELF	    7fecbf40e000-    7fecbf616000	Deferred        libffi.so.6
ELF	    7fecbf616000-    7fecbf830000	Deferred        libresolv.so.2
ELF	    7fecbf830000-    7fecbfa34000	Deferred        libkeyutils.so.1
ELF	    7fecbfa34000-    7fecbfc80000	Deferred        libdbus-1.so.3
ELF	    7fecbfc80000-    7fecbff00000	Deferred        libgmp.so.10
ELF	    7fecbff00000-    7fecc0133000	Deferred        libhogweed.so.4
ELF	    7fecc0133000-    7fecc0368000	Deferred        libnettle.so.6
ELF	    7fecc0368000-    7fecc057b000	Deferred        libtasn1.so.6
ELF	    7fecc057b000-    7fecc07ae000	Deferred        libidn.so.11
ELF	    7fecc07ae000-    7fecc0ade000	Deferred        libp11-kit.so.0
ELF	    7fecc0ade000-    7fecc0ce9000	Deferred        libkrb5support.so.0
ELF	    7fecc0ce9000-    7fecc0eed000	Deferred        libcom_err.so.2
ELF	    7fecc0eed000-    7fecc111f000	Deferred        libk5crypto.so.3
ELF	    7fecc111f000-    7fecc13f2000	Deferred        libkrb5.so.3
ELF	    7fecc13f2000-    7fecc1603000	Deferred        libavahi-client.so.3
ELF	    7fecc1603000-    7fecc180f000	Deferred        libavahi-common.so.3
ELF	    7fecc180f000-    7fecc1ba1000	Deferred        libgnutls.so.30
ELF	    7fecc1ba1000-    7fecc1deb000	Deferred        libgssapi_krb5.so.2
ELF	    7fecc1deb000-    7fecc2075000	Deferred        libcups.so.2
ELF	    7fecc209b000-    7fecc22a1000	Deferred        libxfixes.so.3
ELF	    7fecc22a1000-    7fecc24ab000	Deferred        libxcursor.so.1
ELF	    7fecc24ab000-    7fecc26bb000	Deferred        libxi.so.6
ELF	    7fecc26bb000-    7fecc28be000	Deferred        libxcomposite.so.1
ELF	    7fecc28be000-    7fecc2ac9000	Deferred        libxrandr.so.2
ELF	    7fecc2ac9000-    7fecc2cd3000	Deferred        libxrender.so.1
ELF	    7fecc2cd3000-    7fecc2ed9000	Deferred        libxxf86vm.so.1
ELF	    7fecc2ed9000-    7fecc30dc000	Deferred        libxinerama.so.1
ELF	    7fecc30dc000-    7fecc32e4000	Deferred        librt.so.1
ELF	    7fecc32e4000-    7fecc34f9000	Deferred        libbsd.so.0
ELF	    7fecc34f9000-    7fecc36ff000	Deferred        libxdmcp.so.6
ELF	    7fecc36ff000-    7fecc3903000	Deferred        libxau.so.6
ELF	    7fecc3903000-    7fecc3b2a000	Deferred        libxcb.so.1
ELF	    7fecc3b2a000-    7fecc3e63000	Deferred        libx11.so.6
ELF	    7fecc3e63000-    7fecc4075000	Deferred        libxext.so.6
ELF	    7fecc409b000-    7fecc4339000	Deferred        winex11<elf>
  \-PE	    7fecc40b0000-    7fecc4339000	\               winex11
ELF	    7fecc43c8000-    7fecc45f3000	Deferred        libexpat.so.1
ELF	    7fecc45f3000-    7fecc4836000	Deferred        libfontconfig.so.1
ELF	    7fecc4836000-    7fecc4a68000	Deferred        libpng16.so.16
ELF	    7fecc4a68000-    7fecc4d1b000	Deferred        libfreetype.so.6
ELF	    7fecc4d1b000-    7fecc4f3a000	Deferred        concrt140<elf>
  \-PE	    7fecc4d20000-    7fecc4f3a000	\               concrt140
ELF	    7fecc4f3a000-    7fecc51a0000	Deferred        oleacc<elf>
  \-PE	    7fecc4f40000-    7fecc51a0000	\               oleacc
ELF	    7fecc51a0000-    7fecc53c6000	Deferred        imm32<elf>
  \-PE	    7fecc51b0000-    7fecc53c6000	\               imm32
ELF	    7fecc53c6000-    7fecc5602000	Deferred        uxtheme<elf>
  \-PE	    7fecc53d0000-    7fecc5602000	\               uxtheme
ELF	    7fecc5602000-    7fecc5815000	Deferred        msimg32<elf>
  \-PE	    7fecc5610000-    7fecc5815000	\               msimg32
ELF	    7fecc5815000-    7fecc5b93000	Deferred        oleaut32<elf>
  \-PE	    7fecc5840000-    7fecc5b93000	\               oleaut32
ELF	    7fecc5b93000-    7fecc5e99000	Deferred        msvcr120<elf>
  \-PE	    7fecc5bc0000-    7fecc5e99000	\               msvcr120
ELF	    7fecc5e99000-    7fecc6119000	Deferred        dwrite<elf>
  \-PE	    7fecc5ea0000-    7fecc6119000	\               dwrite
ELF	    7fecc6119000-    7fecc6331000	Deferred        ncrypt<elf>
  \-PE	    7fecc6120000-    7fecc6331000	\               ncrypt
ELF	    7fecc6331000-    7fecc657e000	Deferred        usp10<elf>
  \-PE	    7fecc6340000-    7fecc657e000	\               usp10
ELF	    7fecc657e000-    7fecc6795000	Deferred        dwmapi<elf>
  \-PE	    7fecc6580000-    7fecc6795000	\               dwmapi
ELF	    7fecc6795000-    7fecc6afd000	Deferred        wined3d<elf>
  \-PE	    7fecc67b0000-    7fecc6afd000	\               wined3d
ELF	    7fecc6afd000-    7fecc6d2a000	Deferred        dxgi<elf>
  \-PE	    7fecc6b00000-    7fecc6d2a000	\               dxgi
ELF	    7fecc6d2a000-    7fecc6f41000	Deferred        uiautomationcore<elf>
  \-PE	    7fecc6d30000-    7fecc6f41000	\               uiautomationcore
ELF	    7fecc6f41000-    7fecc716c000	Deferred        mpr<elf>
  \-PE	    7fecc6f50000-    7fecc716c000	\               mpr
ELF	    7fecc716c000-    7fecc73f1000	Deferred        wininet<elf>
  \-PE	    7fecc7180000-    7fecc73f1000	\               wininet
ELF	    7fecc73f1000-    7fecc7638000	Deferred        winspool<elf>
  \-PE	    7fecc7400000-    7fecc7638000	\               winspool
ELF	    7fecc7638000-    7fecc7ac7000	Deferred        shell32<elf>
  \-PE	    7fecc7650000-    7fecc7ac7000	\               shell32
ELF	    7fecc7ac7000-    7fecc7dbe000	Deferred        comdlg32<elf>
  \-PE	    7fecc7ad0000-    7fecc7dbe000	\               comdlg32
ELF	    7fecc7dbe000-    7fecc80ef000	Deferred        comctl32<elf>
  \-PE	    7fecc7dd0000-    7fecc80ef000	\               comctl32
ELF	    7fecc80ef000-    7fecc837c000	Deferred        shlwapi<elf>
  \-PE	    7fecc8100000-    7fecc837c000	\               shlwapi
ELF	    7fecc837c000-    7fecc8599000	Deferred        libz.so.1
ELF	    7fecc8599000-    7fecc8808000	Deferred        dbghelp<elf>
  \-PE	    7fecc85a0000-    7fecc8808000	\               dbghelp
ELF	    7fecc8808000-    7fecc8a35000	Deferred        msacm32<elf>
  \-PE	    7fecc8810000-    7fecc8a35000	\               msacm32
ELF	    7fecc8a35000-    7fecc8cca000	Deferred        rpcrt4<elf>
  \-PE	    7fecc8a50000-    7fecc8cca000	\               rpcrt4
ELF	    7fecc8cca000-    7fecc906b000	Deferred        ole32<elf>
  \-PE	    7fecc8cf0000-    7fecc906b000	\               ole32
ELF	    7fecc906b000-    7fecc932e000	Deferred        winmm<elf>
  \-PE	    7fecc9070000-    7fecc932e000	\               winmm
ELF	    7fecc932e000-    7fecc9546000	Deferred        userenv<elf>
  \-PE	    7fecc9330000-    7fecc9546000	\               userenv
ELF	    7fecc9546000-    7fecc9772000	Deferred        iphlpapi<elf>
  \-PE	    7fecc9550000-    7fecc9772000	\               iphlpapi
ELF	    7fecc9772000-    7fecc998c000	Deferred        version<elf>
  \-PE	    7fecc9780000-    7fecc998c000	\               version
ELF	    7fecc998c000-    7fecc9dc8000	Deferred        user32<elf>
  \-PE	    7fecc99b0000-    7fecc9dc8000	\               user32
ELF	    7fecc9dc8000-    7fecca058000	Deferred        advapi32<elf>
  \-PE	    7fecc9de0000-    7fecca058000	\               advapi32
ELF	    7fecca058000-    7fecca3d3000	Deferred        gdi32<elf>
  \-PE	    7fecca070000-    7fecca3d3000	\               gdi32
ELF	    7fecca3d3000-    7fecca5e6000	Deferred        psapi<elf>
  \-PE	    7fecca3e0000-    7fecca5e6000	\               psapi
ELF	    7fecca5e6000-    7fecca7fa000	Deferred        api-ms-win-crt-conio-l1-1-0<elf>
  \-PE	    7fecca5f0000-    7fecca7fa000	\               api-ms-win-crt-conio-l1-1-0
ELF	    7fecca7fa000-    7feccaa0d000	Deferred        api-ms-win-crt-locale-l1-1-0<elf
PE	    7fecca800000-    7feccaa0d000	Deferred        api-ms-win-crt-locale-l1-1-0
ELF	    7feccaa0d000-    7feccac21000	Deferred        api-ms-win-crt-time-l1-1-0<elf>
  \-PE	    7feccaa10000-    7feccac21000	\               api-ms-win-crt-time-l1-1-0
ELF	    7feccac21000-    7feccae3a000	Deferred        api-ms-win-crt-math-l1-1-0<elf>
  \-PE	    7feccac30000-    7feccae3a000	\               api-ms-win-crt-math-l1-1-0
ELF	    7feccae3a000-    7feccb04e000	Deferred        api-ms-win-crt-utility-l1-1-0<el
PE	    7feccae40000-    7feccb04e000	Deferred        api-ms-win-crt-utility-l1-1-0
PE	    7feccb04e000-    7feccb262000	Deferred        api-ms-win-crt-filesystem-l1-1-0
PE	    7feccb050000-    7feccb262000	Deferred        api-ms-win-crt-filesystem-l1-1-0C:\windows\system32\api-ms-win-crt-filesystem-l1-1-0.dll
PE	    7feccb262000-    7feccb475000	Deferred        api-ms-win-crt-environment-l1-1-
PE	    7feccb270000-    7feccb475000	Deferred        api-ms-win-crt-environment-l1-1-C:\windows\system32\api-ms-win-crt-environment-l1-1-0.dll
ELF	    7feccb475000-    7feccb68a000	Deferred        api-ms-win-crt-convert-l1-1-0<el
PE	    7feccb480000-    7feccb68a000	Deferred        api-ms-win-crt-convert-l1-1-0
ELF	    7feccb68a000-    7feccb89f000	Deferred        api-ms-win-crt-stdio-l1-1-0<elf>
  \-PE	    7feccb690000-    7feccb89f000	\               api-ms-win-crt-stdio-l1-1-0
ELF	    7feccb89f000-    7feccbab3000	Deferred        api-ms-win-crt-heap-l1-1-0<elf>
  \-PE	    7feccb8a0000-    7feccbab3000	\               api-ms-win-crt-heap-l1-1-0
ELF	    7feccbab3000-    7feccbcc9000	Deferred        api-ms-win-crt-string-l1-1-0<elf
PE	    7feccbac0000-    7feccbcc9000	Deferred        api-ms-win-crt-string-l1-1-0
ELF	    7feccbcc9000-    7feccbede000	Deferred        api-ms-win-crt-runtime-l1-1-0<el
PE	    7feccbcd0000-    7feccbede000	Deferred        api-ms-win-crt-runtime-l1-1-0
ELF	    7feccbede000-    7feccc0f4000	Deferred        vcruntime140<elf>
  \-PE	    7feccbee0000-    7feccc0f4000	\               vcruntime140
ELF	    7feccc0f4000-    7feccc407000	Deferred        ucrtbase<elf>
  \-PE	    7feccc120000-    7feccc407000	\               ucrtbase
ELF	    7feccc407000-    7feccc741000	Deferred        msvcp140<elf>
  \-PE	    7feccc440000-    7feccc741000	\               msvcp140
ELF	    7feccc841000-    7feccca82000	Deferred        ws2_32<elf>
  \-PE	    7feccc850000-    7feccca82000	\               ws2_32
ELF	    7fecccb82000-    7fecccd94000	Deferred        libnss_files.so.2
ELF	    7fecccd94000-    7fecccfa0000	Deferred        libnss_nis.so.2
ELF	    7fecccfa0000-    7feccd1ba000	Deferred        libnsl.so.1
ELF	    7feccd1ba000-    7feccd3c3000	Deferred        libnss_compat.so.2
ELF	    7feccd89f000-    7feccdab6000	Deferred        libgcc_s.so.1
ELF	    7feccdab6000-    7feccde0c000	Deferred        libm.so.6
ELF	    7feccde0e000-    7fecce012000	Deferred        libdl.so.2
ELF	    7fecce012000-    7fecce3f2000	Deferred        libc.so.6
ELF	    7fecce3f2000-    7fecce611000	Deferred        libpthread.so.0
ELF	    7fecce637000-    7fecce9dc000	Dwarf           libwine.so.1
ELF	    7fecce9de000-    7feccec08000	Deferred        ld-linux-x86-64.so.2
ELF	    7ffd573c0000-    7ffd573c1000	Deferred        [vdso].so
```

Which makes sense, since my default wineprefix isn't supporting video decoding, unlike my fallout4 wineprefix. However, the fonts not working, even after reinstalling, is strange.

I would expect that Vortex would be able to be ported to Linux semi-decently, since Electron isn't too hard to port for multiple platforms. The biggest issue would be to port the c++ libs used, which from what the dependencies list tells me, involves *"visual c++ build tools 2015"*. I'm unaware of how compatible they would be with g++, though I'd be interested in seeing if we can manage to work something out.

Again, I get it's not within the scope of the alpha, nor would it be high priority, but it's something interesting that I think could spark some discussion.

**Comment by jarrard** [comment]
Works with wine3.7staging BUT when you goto install a mod it brings up a white screen. There is a error output but not sure if I should post it here.

**Comment by TanninOne** [comment]
Vortex in later releases does a check on whether .net is installed and compatible, maybe that gives a hint on what's wrong. Also it seems there is work being done to get Vortex install through Phoenicis (https://github.com/PhoenicisOrg/scripts/issues/903) maybe you could help them as a tester or something?
I'm going to close this because, again, we can't really officially support wine, we just don't have the capacities or knowledge but I'll be available if anyone working on support needs info about Vortex internals.

**Comment by Kreyren** [comment]
@fdrobidoux I don't have this issue on wine-staging-4.4_p1::bobwya with `X alsa esync fontconfig jpeg mono mp3 opencl opengl osmesa png pulseaudio realtime run-exes sdl2 ssl threads truetype udev vkd3d vulkan xcomposite` useflags on gentoo strata using bedrock linux 0.7.2 Poki with custom kernel 5.0.3 with Colin Kolivas and Architecture optimization patches.
![image](https://user-images.githubusercontent.com/11302521/55412690-35dac400-5568-11e9-87ac-c84f69f6f54e.png)

### WORKAROUNDS
1) `dxgi.dll` from DXVK is mandatory for vortex to start on Linux, i believe that using said dep should solve the issue you are facing now.
2) .NET for fomods which requires dotnet46 and higger with libs referenced on https://github.com/PhoenicisOrg/scripts/issues/903 to be parsed in `$WINEPREFIX/drive_c/windows/Microsoft.NET/Framework64/foo.dll`.  Currently **HELP WANTED** due licencing issue on phoenicis.
3) Download location has to be set to accept linux hierarchy.
4) Scan function in vortex needs `readline` apparently which i didn't make it work yet.

**Comment by fdrobidoux** [comment]
It's been a while. I don't have a Ubuntu instance anymore that's running the desktop environment, so I can't confirm if it's working or not, neither do I have any remaining interest in Vortex or game modding as a whole.

I'm glad its running on your end, but meanwhile I moved on.

Have a nice day regardless !

**Comment by jarrard** [comment]
I mostly mess with this stuff as a small interest to see how things are progressing under Linux with Bethesda games and modding in general. 

In actual fact when I want to really play a heavily modded game, I'm on windows10 boot, unfortunately Linux is just no there yet, but it can be made to work but the effort (due to issues and workarounds) really saps all the fun out of it....

Whats needed is a NATIVE Linux Mod Organizer 2 or Vortex which uses NATIVE linux libraries to mount virtual file systems, MO2 VFS under windows10 is insanely fast to mount and launch the game, THAT is what Linux needs. ATM you have the opposite.

**Comment by Kreyren** [comment]
> unfortunately Linux is just no there yet,

+- 90% of mods are working for me, but agree it's pita to configure

> Whats needed is a NATIVE Linux Mod Organizer 2 or Vortex which uses NATIVE linux libraries to mount virtual file systems, MO2 VFS under windows10 is insanely fast to mount and launch the game, THAT is what Linux needs. ATM you have the opposite.

Referencing: https://github.com/PhoenicisOrg/scripts/issues/919 if you find a way to implement this effectively in a way where engine(wine), engine patches and changes to game can be sandboxed(?) in a way that makes everything modular meaning to be removed/installed on demand then let me know and i do my best to implement it.

**Comment by jarrard** [comment]
Why mount a vfs through wine at all? I said native, then just pass game execution to linux steam and let steamplay do the rest (renaming skse to *Launcher.exe does allows this to work).

There really isn't a need for wine to get involved when it comes to installing and mounting mod files for a bethesda game.

**Comment by Kreyren** [comment]
@jarrard we can't rely on steamplay for everything since it's just wine with additional patches targeted to work with steam games only -> totally different usecase and it's much more effective to use upstream wine in general if you know how to use it..

example: you can't really use steamplay for league of legends since it currently needs patched glibc and patched wine (https://bugs.winehq.org/show_bug.cgi?id=47198) + DXVK through dgvoodoo wrapper or DXV9 to get performance supperior to native.

How do you deal with this scenario assuming that we can't distribute wine for each game and modification and it's modification combination? 

Example skyrim having 55061 mods and assuming that 87% works without pita changes -> 47903.07 multiplied by number of workarounds (verbs) in phoenicis on https://github.com/PhoenicisOrg/scripts/tree/master/Engines/Wine/Verbs which is 49 multiplied by lets say 2860 wine patches and dependency patches and configuration (probably under shot) you get around 6 713 136 229 possible configurations.. assuming that phoenicis won't compile wine from source so ideally we want to have one wine with verbs that can be applied on demand (no problem atm) and patches which is problematic since they require compiling wine usually and making changes in wine itself so it's hard to redo them since we need to change them depending on what mods are used for compatibility.

> There really isn't a need for wine to get involved when it comes to installing and mounting mod files for a bethesda game.

How do you mount them so that they can be enabled/disabled on demand? Just vfs each mod and apply changes to gamedir?

**Comment by Kreyren** [comment]
like compiling wine would solve all issues with wine which is not accepted in phoenicis..

For mods we need method to enable/disable them and i'm not sure if vfs is the best way to do this

**Comment by jarrard** [comment]
MO2 is IMO the best way for windows10, I do not know what VFS systems work well under Linux that could be compared to, I just know that raw symlinks (via script) seem to cause significant slow downs and takes longer to mount. 

Sure once the games all loaded up its mostly fine, but what if you want to do mod testing and need to load and start the game 100x in a row? that 5mins boot time is a nightmare... I can only assume the symlink method causes this, I haven't actually tried directly copying the mods to my skyrimse folder because that is a messy nightmare.

**Comment by Kreyren** [comment]
> MO2 is IMO the best way for windows10,

Would agree that ModOrganizer2 is best way to handle mods in general.

> Sure once the games all loaded up its mostly fine, but what if you want to do mod testing and need to load and start the game 100x in a row?

then you load it 300 times make custom patches and stage 4700 under 248 mods in skyrim to make it work and rage quit when one thing breaks since it means that you have to test it 300 times again.. from personall experience with skyrim 

SOLUTION_01: i was thinking that we can make phoenicis website with forums and stuff to organize crowdsource to handle compatibility per engine(wine), engine patches, workarounds, hardware, software and configuration used..

>  I just know that raw symlinks (via script) seem to cause significant slow downs and takes longer to mount.

Doubt it beeing true assuming that we are using same files.. but it's very likely to cause breakage depending on the implementation.

Afaik symlinks ware issue in wine long time ago which would explain slowdowns or there might be issue in current implementation.. **reference needed**

Assuming this issue beeing present only on windows it's irelevant since windows is too unreliable.

> If MO2 VFS can be made to work via wine, then that be great but I have serious doubts...

i think @TanninOne told me what is beeing used, but i forgett >.>.. Can try to make it work on linux current issue tracker mensions `NtQueryDirectoryFileEx` conflict only https://bugs.winehq.org/show_bug.cgi?id=46697

**Comment by Kreyren** [comment]
Assuming blocked by https://bugs.winehq.org/show_bug.cgi?id=46697
